src/test/java/org/apache/datasketches/quantilescommon/KolmogorovSmirnovTest.java

```diff
@@ -0,0 +1,357 @@
+     s1.update(x + 500);
+     s2.update(x);
+   }
+   final double delta = KllSketch.getNormalizedRankError(k, false);
+   println("D = " + KolmogorovSmirnov.computeKSDelta(s1, s2));
+   assertEquals(KolmogorovSmirnov.computeKSDelta(s1, s2), 1.0, delta);
+ }
+
+ @Test
+ public void checkDisjointDistributionKllFloats() {
+   final int k = 256;
+   final KllFloatsSketch s1 = KllFloatsSketch.newHeapInstance(k);
+   final KllFloatsSketch s2 = KllFloatsSketch.newHeapInstance(k);
+
+   final Random rand = new Random(1);
+
+   final int n =  (3 * k) - 1;
+   for (int i = 0; i < n; ++i) {
+     final float x = (float)rand.nextGaussian();
+     s1.update(x + 500);
+     s2.update(x);
+   }
+   final double delta = KllSketch.getNormalizedRankError(k, false);
+   println("D = " + KolmogorovSmirnov.computeKSDelta(s1, s2));
+   assertEquals(KolmogorovSmirnov.computeKSDelta(s1, s2), 1.0, delta);
+ }
+
+ @Test
+ public void checkIdenticalDistributionClassicDoubles() {
+   final int k = 256;
+   final UpdateDoublesSketch s1 = DoublesSketch.builder().setK(k).build();
+
+   final Random rand = new Random(1);
+
+   final int n =  (3 * k) - 1;
+   for (int i = 0; i < n; ++i) {
+     final double x = rand.nextGaussian();
+     s1.update(x);
+   }
+   println("D = " + KolmogorovSmirnov.computeKSDelta(s1, s1));
+   assertEquals(KolmogorovSmirnov.computeKSDelta(s1, s1), 0.0, 0.0);
+ }
+
+ @Test
+ public void checkIdenticalDistributionKllDoubles() {
+   final int k = 256;
+   final KllDoublesSketch s1 = KllDoublesSketch.newHeapInstance(k);
+
+   final Random rand = new Random(1);
+
+   final int n =  (3 * k) - 1;
+   for (int i = 0; i < n; ++i) {
+     final double x = rand.nextGaussian();
+     s1.update(x);
+   }
+   println("D = " + KolmogorovSmirnov.computeKSDelta(s1, s1));
+   assertEquals(KolmogorovSmirnov.computeKSDelta(s1, s1), 0.0, 0.0);
+ }
+
+ @Test
+ public void checkIdenticalDistributionKllFloats() {
+   final int k = 256;
+   final KllFloatsSketch s1 = KllFloatsSketch.newHeapInstance(k);
+
+   final Random rand = new Random(1);
+
+   final int n =  (3 * k) - 1;
+   for (int i = 0; i < n; ++i) {
+     final float x = (float)rand.nextGaussian();
+     s1.update(x);
+   }
+   println("D = " + KolmogorovSmirnov.computeKSDelta(s1, s1));
+   assertEquals(KolmogorovSmirnov.computeKSDelta(s1, s1), 0.0, 0.0);
+ }
+
+ @Test
+ public void checkSameDistributionDifferentClassicDoublesSketches() {
+   final int k = 256;
+   final UpdateDoublesSketch s1 = DoublesSketch.builder().setK(k).build();
+   final UpdateDoublesSketch s2 = DoublesSketch.builder().setK(k).build();
+
+   final Random rand = new Random(1);
+
+   final int n =  (3 * k) - 1;
+   for (int i = 0; i < n; ++i) {
+     final double x = rand.nextGaussian();
+     s1.update(x);
+     s2.update(x);
+   }
+   final double delta = DoublesSketch.getNormalizedRankError(k, false);
+   println("D = " + KolmogorovSmirnov.computeKSDelta(s1, s2));
+   assertEquals(KolmogorovSmirnov.computeKSDelta(s1, s2), 0, delta);
```

Comment:
Here, `delta` is too restrictive in general. It may be fine with the given seed, but conceptually it's the wrong thing to test against. The allowable margin is larger than the single-sketch delta.

Reading test code is a good way to learn how things are supposed to work, so we shouldn't have our tests suggest that this is a good target to check against. Or at the very least a comment noting that we're using it despite it being too tight a bound for this test in general.